File: hackerrank/easy/0022-beautiful-days-at-the-movies/python/solution.py
## solution.py

def beautiful_days(i: int, j: int, k: int) -> int:
    def reverse_number(num: int) -> int:
        return int(str(num)[::-1])

    count = 0
    for day in range(i, j + 1):
        if abs(day - reverse_number(day)) % k == 0:
            count += 1
    return count


# --- demo run ---
if __name__ == "__main__":
    print(beautiful_days(20, 23, 6))  # 2
    print(beautiful_days(10, 13, 3))  # 2

File: hackerrank/easy/0022-beautiful-days-at-the-movies/swift/solution.swift
// solution.swift

func beautifulDays(i: Int, j: Int, k: Int) -> Int {
    func reverseNumber(_ num: Int) -> Int {
        var n = num
        var reversed = 0
        while n > 0 {
            reversed = reversed * 10 + n % 10
            n /= 10
        }
        return reversed
    }
    
    var count = 0
    for day in i...j {
        if abs(day - reverseNumber(day)) % k == 0 {
            count += 1
        }
    }
    return count
}

// --- demo run ---
print(beautifulDays(i: 20, j: 23, k: 6)) // 2
print(beautifulDays(i: 10, j: 13, k: 3)) // 2